src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs

```diff
@@ -226,6 +226,142 @@ public static void BuildChainExtraStoreUntrustedRoot()
             }
         }
 
+        // Linux and Windows do not search the default system stores with CustomRootTrust is enabled
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Linux | TestPlatforms.Windows)]
+        public static void SystemTrustCertificateWithCustomTrustStore()
+        {
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var testCert = new X509Certificate2(Path.Combine("TestData", "test.pfx"), TestData.ChainPfxPassword))
+            using (var chainHolder = new ChainHolder())
+            {
+                X509Chain chain = chainHolder.Chain;
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                chain.ChainPolicy.VerificationTime = microsoftDotCom.NotBefore.AddSeconds(1);
+                chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
+                chain.ChainPolicy.CustomTrustStore.Add(testCert);
+
+                Assert.False(chain.Build(microsoftDotCom));
+                Assert.Equal(2, chain.ChainElements.Count);
+                Assert.Equal(X509ChainStatusFlags.PartialChain, chain.AllStatusFlags());
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.OSX)]
+        public static void SystemTrustCertificateWithCustomTrustStoreOSX()
+        {
+            using (var microsoftDotCom = new X509Certificate2(TestData.MicrosoftDotComSslCertBytes))
+            using (var testCert = new X509Certificate2(Path.Combine("TestData", "test.pfx"), TestData.ChainPfxPassword))
+            using (var chainHolder = new ChainHolder())
+            {
+                X509Chain chain = chainHolder.Chain;
+                chain.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
+                chain.ChainPolicy.VerificationTime = microsoftDotCom.NotBefore.AddSeconds(1);
+                chain.ChainPolicy.TrustMode = X509ChainTrustMode.CustomRootTrust;
+                chain.ChainPolicy.CustomTrustStore.Add(testCert);
+
+                Assert.False(chain.Build(microsoftDotCom));
+                Assert.Equal(3, chain.ChainElements.Count);
+                Assert.Equal(X509ChainStatusFlags.UntrustedRoot, chain.AllStatusFlags());
```

Comment:
Since this and the previous test only differ by the last two lines, please merge them (just branch the test internally using `if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))`)